File: VB/EventToCommandBehaviorParametersConverters/PassingEventArgumentsToCommandsASISUserControl.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Utils.MVVM
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Controls
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Namespace DxSampleEventToCommandBehaviorParametersConverters
    Partial Public Class PassingEventArgumentsToCommandsASISUserControl
        Inherits DevExpress.XtraEditors.XtraUserControl
        Public Sub New() 
            InitializeComponent()
            Dim mvvmContext As New MVVMContext()
            mvvmContext.ContainerControl = Me
        
            Dim panel As New PanelControl()
            panel.Dock = DockStyle.Top
            panel.Parent = Me
        
            Dim label As New LabelControl()
            label.Text = "Click to Execute Command"
            label.Dock = DockStyle.Fill
            label.AutoSizeMode = LabelAutoSizeMode.None
            label.Appearance.TextOptions.HAlignment = DevExpress.Utils.HorzAlignment.Center
            label.Appearance.TextOptions.VAlignment = DevExpress.Utils.VertAlignment.Center
            label.Parent = panel
        

            mvvmContext.ViewModelType = GetType(MouseDownAwareViewModel)
            ' UI binding for the EventToCommand behavior
            mvvmContext.OfType(Of MouseDownAwareViewModel)().WithEvent(Of MouseEventArgs)(label, "MouseDown").
                EventToCommand(Sub(x) x.ReportArgs(New MouseEventArgs(MouseButtons.Left, 0, 0, 0, 0)))
        End Sub
         Private Sub OnDisposing()
            Dim context = MVVMContext.FromControl(Me)
            If context IsNot Nothing Then
                context.Dispose()
            End If
        End Sub
        Public Class MouseDownAwareViewModel
            Public Sub New()
                Message = "MouseDown performed!"
            End Sub
            Public Property Message() As String
            Protected ReadOnly Property MessageBoxService() As IMessageBoxService
                Get
                    Return Me.GetService(Of IMessageBoxService)()
                End Get
            End Property
            Public Sub Report(ByVal message As String)
                MessageBoxService.ShowMessage(message)
            End Sub
            Public Sub ReportArgs(ByVal args As MouseEventArgs)
        
                Dim msg As String = String.Join(", ",
                                          "Button: " & args.Button.ToString(),
                                          "Location: " & args.Location.ToString(),
                                          "Clicks: " & args.Clicks.ToString(),
                                          "Delta: " & args.Delta.ToString())
                MessageBoxService.ShowMessage("Args = {" & msg & "}")
            End Sub
            Public Sub ReportLocation(ByVal pt As Point)
                MessageBoxService.ShowMessage("Location = " & pt.ToString())
            End Sub
        End Class
    	End Class
End Namespace

File: CS/EventToCommandBehaviorParametersConverters/PassingEventArgumentsToCommandsASISUserControl.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.MVVM;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using System;
using System.Drawing;
using System.Windows.Forms;

namespace DxSampleEventToCommandBehaviorParametersConverters {
    public partial class PassingEventArgumentsToCommandsASISUserControl : XtraUserControl {
        public PassingEventArgumentsToCommandsASISUserControl() {
            InitializeComponent();
            #region SetUp
            MVVMContext mvvmContext = new MVVMContext();
            mvvmContext.ContainerControl = this;
        
            PanelControl panel = new PanelControl();
            panel.Dock = DockStyle.Top;
            panel.Parent = this;
        
            LabelControl label = new LabelControl();
            label.Text = "Click to Execute Command";
            label.Dock = DockStyle.Fill;
            label.AutoSizeMode = LabelAutoSizeMode.None;
            label.Appearance.TextOptions.HAlignment = DevExpress.Utils.HorzAlignment.Center;
            label.Appearance.TextOptions.VAlignment = DevExpress.Utils.VertAlignment.Center;
            label.Parent = panel;
        
            #endregion SetUp

            #region #passingEventArgumentsToCommandsASIS
            mvvmContext.ViewModelType = typeof(MouseDownAwareViewModel);
            // UI binding for the EventToCommand behavior
            mvvmContext.OfType<MouseDownAwareViewModel>()
                .WithEvent<MouseEventArgs>(label, "MouseDown")
                .EventToCommand(x => x.ReportArgs(new MouseEventArgs(MouseButtons.Left, 0, 0, 0, 0)));
            #endregion #passingEventArgumentsToCommandsASIS
        }
        #region CleanUp
        void OnDisposing() {
            var context = MVVMContext.FromControl(this);
            if(context != null) context.Dispose();
        }
        #endregion CleanUp
        public class MouseDownAwareViewModel {
            public MouseDownAwareViewModel() {
                Message = "MouseDown performed!";
            }
            public string Message { get; set; }
            protected IMessageBoxService MessageBoxService {
                get { return this.GetService<IMessageBoxService>(); }
            }
            public void Report(string message) {
                MessageBoxService.ShowMessage(message);
            }
            public void ReportArgs(MouseEventArgs args) {
                string message = string.Join(", ",
                        "Button: " + args.Button.ToString(),
                        "Location: " + args.Location.ToString(),
                        "Clicks: " + args.Clicks.ToString(),
                        "Delta: " + args.Delta.ToString());
                MessageBoxService.ShowMessage("Args = {" + message + "}");
            }
            public void ReportLocation(Point pt) {
                MessageBoxService.ShowMessage("Location = " + pt.ToString());
            }
        }
    }
}